p2p/protocol/identify/id.go

```diff
@@ -408,7 +414,9 @@ func (ids *IDService) sendIdentifyResp(s network.Stream) {
 	log.Debugf("%s sent message to %s %s", ID, c.RemotePeer(), c.RemoteMultiaddr())
 }
 
-func (ids *IDService) handleIdentifyResponse(s network.Stream) {
+func (ids *IDService) handleIdentifyResponse(s network.Stream) error {
```

Comment:
why are we returning an error now? There isn't much we can do about it and we appear to always ignore it.

Reply:
@Stebalien 

This is to ensure that we publish an `IdentifyCompleted` only if the stream read was successful. If the stream read fails, say because of a timeout, we want to publish an `EvtPeerIdentificationFailed` event.

Reply:
Ah, I see. We should be returning an error from consumeMessage in that case.